components/movementsensor/gpsutils/vrs.go

```diff
@@ -3,32 +3,46 @@ package gpsutils
 
 import (
 	"bufio"
+	"context"
 	"encoding/base64"
 	"fmt"
 	"net"
 	"net/url"
+	"sync"
+	"time"
 
 	"go.viam.com/rdk/logging"
 )
 
+// VRS contains the VRS.
+type VRS struct {
+	ntripInfo               *NtripInfo
+	ReaderWriter            *bufio.ReadWriter
+	Conn                    net.Conn
+	activeBackgroundWorkers sync.WaitGroup
+	cancelCtx               context.Context
+	cancelFunc              func()
```

Comment:
Sigh. This is the antipattern I've been trying to get rid of in my 20% time. Remove these 3 fields, and use a `utils.StoppableWorkers` object instead.

```go
type VRS struct {
    ...
    workers utils.StoppableWorkers
}

func (vrs *VRS) StartGGAThread(ggaFunc func() (string, error)) {
    if vrs.workers != nil {
        // Is it okay to start a second thread like this? If not, return errors.
        // If so, use vrs.workers.AddWorkers below, instead of utils.NewStoppableWorkers.
    }

    vrs.workers = utils.NewStoppableWorkers(func(cancelCtx context.Context) {
        ...
    })
}

func (vrs *VRS) Close() error {
    vrs.workers.Stop()
    return vrs.Conn.Close()
}
```